src/test/scala/com/raquo/airstream/split/SplittableTypeSpec.scala

```diff
@@ -0,0 +1,474 @@
+package com.raquo.airstream.split
+
+import com.raquo.airstream.UnitSpec
+import com.raquo.airstream.eventbus.EventBus
+import com.raquo.airstream.fixtures.{Effect, TestableOwner}
+import com.raquo.airstream.state.Var
+import com.raquo.airstream.split.SplittableTypeMacros.{splitMatch, handleCase, toSignal, toStream}
+
+import scala.collection.{immutable, mutable}
+import scala.scalajs.js
+import com.raquo.airstream.ShouldSyntax.shouldBeEmpty
+
+class SplittableTypeSpec extends UnitSpec {
+
+  sealed trait Foo
+
+  final case class Bar(strOpt: Option[String]) extends Foo
+  enum Baz extends Foo {
+    case Baz1, Baz2
+  }
+  case object Tar extends Foo
+
+  final case class Res(result: Any)
+  
+  it("split match signal") {
+    val effects = mutable.Buffer[Effect[String]]()
+
+    val myVar = Var[Foo](Bar(Some("initial")))
+
+    val owner = new TestableOwner
+
+    val signal = myVar.signal
+      .splitMatch
```

Comment:
Since this test file uses the new Scala 3 only features, the test file itself should be Scala 3 only